src/lib.rs

```diff
@@ -0,0 +1,341 @@
+extern crate chrono;
+extern crate oauth2;
+extern crate reqwest;
+extern crate url;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+extern crate tiny_http;
+#[macro_use]
+extern crate failure;
+
+use chrono::NaiveDate;
+use oauth2::{AuthType, Config as OAuth2Config};
+use reqwest::header::{Authorization, Bearer, Headers, UserAgent};
+use reqwest::Method;
+
+pub mod errors;
+
+use errors::Error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Token(oauth2::Token);
+
+pub struct FitbitClient {
+    client: reqwest::Client,
+    base: url::Url,
+}
+
+pub enum DateQuery {
+    ForDate(NaiveDate),
+    PeriodicSince(NaiveDate, Period),
+    Range(NaiveDate, NaiveDate),
+}
+
+/// UserProfile is a partial serialization struct of the Fitbit API profile. See:
+/// https://dev.fitbit.com/build/reference/web-api/user/
+#[derive(Serialize, Deserialize, Debug)]
+pub struct UserProfile {
+    age: i64,
+    #[serde(rename = "offsetFromUTCMillis")]
+    utc_offset: i64,
+    // TODO: lots of fields...
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct UserProfileResult {
```

Comment:
The stuct above makes sense to me; I'm not sure I'm seeing the reason for wrapping into another struct.

Reply:
This is a mapping of their nested response
```
{
  "user": { tons of fileds }
}
```
Serde wants each object to be mapped precisely. We could avoid this by deserializing something like `HashMap<String, UserProfile>`, and looking up the sub-object.

Either way, they have a lot of objects with this shape in Fitbit. I'd be down to hide the wrapper types from the public api, come to think of it, although we might need them for deserialization. On the other hand, future library users might expect us to map the Json response faithfully - even if it's kind of cludgy.